driver-sync/src/main/com/mongodb/client/vault/KeyVault.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.client.vault;
+
+import com.mongodb.client.model.vault.DataKeyOptions;
+import com.mongodb.client.model.vault.EncryptOptions;
+import org.bson.BsonBinary;
+import org.bson.BsonValue;
+
+import java.io.Closeable;
+
+/**
+ * The Key vault.
+ * <p>
+ * Used to create data encryption keys, and to explicitly encrypt and decrypt values when auto-encryption is not an option.
+ *
+ * @since 3.11
+ */
+public interface KeyVault extends Closeable {
+
+    /**
+     * Create a data key with the given KMS provider.
+     *
+     * @param kmsProvider the KMS provider
+     * @return the identifier for the data key
+     */
+    BsonBinary createDataKey(String kmsProvider);
+
+    /**
+     * Create a data key with the given KMS provider and options.
+     *
+     * @param kmsProvider    the KMS provider
+     * @param dataKeyOptions the options for data key creation
+     * @return the identifier for the data key
+     */
+    BsonBinary createDataKey(String kmsProvider, DataKeyOptions dataKeyOptions);
+
+    // TODO should these be generic methods that can use a codec registry
```

Comment:
Does this TODO still stand?

Reply:
I think it does.  BsonValue isn't the most friendly but on the other hand most users are unaware of the registry.  Could default to the default registry, or configure the registry in the settings.  What do you think?